runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -978,6 +981,111 @@ doubleMaxMinHelper(TR::Node *node, TR::CodeGenerator *cg, bool isMaxOp)
+   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_MASK8, node, equalRegion);
+
+   // If first operand is NaN, then we are done, otherwise fallthrough to move second operand as result
+   generateRXEInstruction(cg, node->getOpCode().isDouble() ? TR::InstOpCode::TCDB : TR::InstOpCode::TCEB, node, lhsReg, generateS390MemoryReference(0x00F, cg),0);
+   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_MASK4, node, cFlowRegionEnd);
+   //branch to swap label, since either second operand is NaN, or entire satisfies the alternate condition code
+   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, swap);
+
+   //code for handling +0/-0 comparisons when operands are equal
+   generateS390LabelInstruction(cg, TR::InstOpCode::label, node, equalRegion);
+   if (node->getOpCode().isMax())
+      {
+      //For Max calls, checking if first operand is +0, then we are done, otherwise fall through for swap
+      generateRXEInstruction(cg, node->getOpCode().isDouble() ? TR::InstOpCode::TCDB : TR::InstOpCode::TCEB, node, lhsReg, generateS390MemoryReference(0x800, cg), 0);  // lhsReg is +0 ?
+      generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_MASK4, node, cFlowRegionEnd);   // it is +0
+      }
+   else if (node->getOpCode().isMin())
+      {
+      //For Min calls, checking if first operand is not +0, then we are done, otherwise fall through for swap
+      generateRXEInstruction(cg, node->getOpCode().isDouble() ? TR::InstOpCode::TCDB : TR::InstOpCode::TCEB, node, lhsReg, generateS390MemoryReference(0x400, cg), 0);  // lhsReg is -0 ?
+      generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_MASK4, node, cFlowRegionEnd);
+      }
+
+   generateS390LabelInstruction(cg, TR::InstOpCode::label, node, swap);
+   //Move resulting operand to lhsReg as fallthrough for alternate Condition Code
+   generateRREInstruction(cg, moveRROp, node, lhsReg, rhsReg);
+
+   TR::RegisterDependencyConditions* deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 2, cg);
+   deps->addPostConditionIfNotAlreadyInserted(lhsReg, TR::RealRegister::AssignAny);
+   deps->addPostConditionIfNotAlreadyInserted(rhsReg, TR::RealRegister::AssignAny);
+
+   generateS390LabelInstruction(cg, TR::InstOpCode::label, node, cFlowRegionEnd, deps);
+   cFlowRegionEnd->setEndInternalControlFlow();
+
+   node->setRegister(lhsReg);
+   cg->decReferenceCount(lhsNode);
+   cg->decReferenceCount(rhsNode);
+
+   return lhsReg;
+   }
+
+TR::Register*
+J9::Z::TreeEvaluator::fminEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   if (cg->getSupportsVectorRegisters())
+      {
+      return fpMinMaxVectorHelper(node, cg, TR::DataTypes::Float, false);
+      }
+   return fpMinMaxHelper(node, cg, TR::InstOpCode::CEBR, TR::InstOpCode::COND_MASK4, TR::InstOpCode::LER);
```

Comment:
Do we need to pass in this information ? Why can't we do same as vector max min helper ?